haystack/preview/components/routers/metadata_router.py

```diff
@@ -0,0 +1,67 @@
+from typing import Any, Dict, List
+
+from haystack.preview import component, default_from_dict, default_to_dict, Document
+from haystack.preview.utils.filters import document_matches_filter
+
+
+@component
+class MetadataRouter:
+    """
+    A component that routes documents to different edges based on their metadata and specified rules.
```

Comment:
Let's always call them connections rather than edges. Also, maybe we can be a bit more generic about what the class can do

```suggestion
    A component that routes documents to different connections based on the content of their fields.
```